### PR TITLE
[test] complete cmake coverage

### PR DESCRIPTION
**Pull request by barendgehrels**
This makes CMake coverage complete for unit tests.
The CMakeLists are generated automatically, and manually edited where indicated.
There is one test added to the Jam suite as well, as indicated.

### CODE REVIEW

File: test/formulas/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    inverse
+    direct_accuracy
+    direct_meridian
+    intersection
+    vertex_longitude
+  )
+  boost_geometry_add_unit_test("formulas" ${item})
+endforeach()
+
+if (NOT APPLE)
```
Comment (barendgehrels) [inline test/formulas/CMakeLists.txt:17]: Manual edit

File: test/geometries/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+foreach(item IN ITEMS
+    adapted
+    boost_array_as_point
+    boost_fusion
+    # boost_polygon
+    boost_range
+    boost_tuple
+    box
+    custom_linestring
+    # initialization
```
Comment (barendgehrels) [inline test/geometries/CMakeLists.txt:18]: The two commented tests would need extra dependencies (for test only).

File: test/io/Jamfile [hunk 3]
```diff
@@ -13,5 +13,6 @@
 # Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
 # http://www.boost.org/LICENSE_1_0.txt)
 
-build-project wkt ; 
+build-project dsv ;
```
Comment (barendgehrels) [inline test/io/Jamfile:16]: The `dsv` Jamfile was there, but not run. For CMake it runs without problems - so I added it for Jam too

File: test/iterators/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+# Copyright (c) 2024 Barend Gehrels, Amsterdam, the Netherlands.
+# Use, modification and distribution is subject to the Boost Software License,
+# Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+# http://www.boost.org/LICENSE_1_0.txt)
+
+foreach(item IN ITEMS
+    closing_iterator
+    # concatenate_iterator
+    ever_circling_iterator
+    # flatten_iterator
```
Comment (barendgehrels) [inline test/iterators/CMakeLists.txt:11]: Two manual edits, needing extra dependencies

File: test/views/CMakeLists.txt [hunk 5]
```diff
@@ -4,13 +4,14 @@
...
+foreach(item IN ITEMS
+    box_view
+    closeable_view
+    enumerate_view
+    random_access_view
+    reversible_closeable
+    reversible_view
+    segment_view
+  )
+  boost_geometry_add_unit_test("views" ${item})
```
Comment (barendgehrels) [inline test/views/CMakeLists.txt:16]: Format change only

**Review by vissarion — APPROVED** [review]
Thanks, it is great to have full cmake support.

There is also a related cmake issue https://github.com/boostorg/geometry/issues/1282 but that could be handled in different PR.

**Pull request merged**